app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts

```diff
@@ -1,28 +1,43 @@
-import { RpcEntry } from '../../types/rpc.js'
 import { EthereumJsonRpcRequest, JsonRpcResponse } from '../../types/JsonRpc-types.js'
 import { JsonRpcResponseError } from '../../utils/errors.js'
 import { serialize } from '../../types/wire-types.js'
 import { keccak256, toUtf8Bytes } from 'ethers'
 import { fetchWithTimeout } from '../../utils/requests.js'
 import { Future } from '../../utils/future.js'
+import { RpcEntry } from '../../types/rpc.js'
+import { getRpcList } from '../../background/storageVariables.js'
 
 type ResolvedResponse = { responseState: 'failed', response: Response } | { responseState: 'success', response: unknown }
 
 export type IEthereumJSONRpcRequestHandler = Pick<EthereumJSONRpcRequestHandler, keyof EthereumJSONRpcRequestHandler>
 export class EthereumJSONRpcRequestHandler {
 	private nextRequestId = 1
-	private rpcEntry: RpcEntry
 	private caching: boolean
 	private pendingCache: Map<string, Future<ResolvedResponse>>
 	private cache: Map<string, ResolvedResponse>
+	private rpcUrl: string
+	private cachedRpcEntry: RpcEntry | undefined = undefined
 
-	constructor(rpcEntry: RpcEntry, caching = false) {
-		this.rpcEntry = rpcEntry
+	constructor(rpcUrl: string, caching = false) {
+		this.rpcUrl = rpcUrl
 		this.caching = caching
 		this.cache = new Map()
 		this.pendingCache = new Map()
-    }
-	public readonly getRpcEntry = () => this.rpcEntry
+		this.getRpcEntryFromStorage()
+	}
+
+
+	private getRpcEntryFromStorage = async () => {
```

Comment:
I don't think we should retrieve the entry from the storage. This class doesn't need the entry and it should work without it

Reply:
ah sorry. This class needs it, and it should take it as input. I don't think the API of this class should change